### ISSUE TITLE
Balancing Proposal

### ISSUE THREAD

**Issue by Dumpling00** [Cam11062/Strange-Berries#9]
Hello! Big fan of your mod, been using it for months now alongside a bunch of friends in our own little server.

We were talking about it the other day, and thought the berries might be a little too easy to overfarm - It becomes really easy to get full stacks of berries and, even though you can't eat multiple at once, it's really just as easy as eating another every time the effects of the previous one dissipate, in order to be with essentially a permanent buff.

This is especially true when the modpack has other mods that enhance your inventory space, like backpacks.

We were thinking, if at all possible, this could be mitigated by restricting players to only eat x amount of berries per day (Configurable?) before throwing up and feeling sick / getting damage. This would also encourage strategical use of the berries.

Main reason we suggest this is because we find it deeply undermines the whole brewing process. No much point in going through the trouble to brew three invisibility potions (for example) that will grant you up to 9 minutes of invisibility, when you can simply carry a stack of 64 invisibility berries (or more) and eat one every 10 seconds. This is a bit clunkier, sure, but when weighted with the cost of brewing and the fact that potions can't stack, it ultimately becomes the better way to consume berries and benefit from their effects.

This is why we think allowing the player to put a cap on how many berries can be consumed daily before getting sick (other than eating multiple berries while other effects are still applied) could solve this issue and entice more people toward the brewing route.

Thank you for your time and consideration!

**Comment by reiieva** [comment]
Hi there! I greatly appreciate your support by using my mod and it's heartwarming to hear that you use it on your own server with your friends. 

I totally agree with you when it comes to the berries being very overpowered and unbalanced, it's something I've always thought is a major issue with my mod, however, no one has really commented on it until now so it's problem I've left unattended to ever since I implemented growing/farming mechanics to the berries so I'm glad you've brought this up and reminded me of this issue. 

About sorting this problem, your idea of adding a cap to how many berries can be consumed daily before getting sick is a great idea and I'm a big fan of it, making this configurable is also very appealing to me and I think this would be a great way to not only as you mentioned balance the use of these berries but also allow for more customisability for anyone using my mod.

Although, implementing said feature at this time may be difficult for me. I don't want to go into too much detail about my life to avoid boring you but I'm currently studying multiple subjects at college whilst working a part time job during much of my free time spent not studying. Because of this, I've said to myself that I would take time away from working on projects as a whole. Although many people would argue that Strange Berries is a small mod (including myself), I'm no experienced programmer, my knowledge is what I would call quite basic and programming is only a side hobby in my life, in my eyes it's pure luck and coincidence that my mod has grown in popularity over the years. Therefore, my point is that it would likely take me some time to work on this idea and my priority right now would be to update the mod to 1.20.5 and 1.20.6 before considering working on this. 

So, to summarise, I really like this change you have proposed with implementing a cap to how many berries can be consumed daily. I would also love to make the amount configurable to allow customisability. However, due to personal reasons, I may not be able to work on this for a while, my priority is to update the mod to 1.20.5 and 1.20.6 then I could consider about working on this. 

I am generous for your support and the time you have taken to write this proposal and I hope you understand my situation.
Whether I will work on this, I can't confirm nor deny, I don't want to make any promises I can't deliver on, but I will leave this issue open and keep you updated when I can.

Kind regards,
Cameron

**Comment by Dumpling00** [comment]
Totally understandable and absolutely no problem! Please take all the time you need. I wish you all the best and all the good luck.
We will be waiting for you! Glad you like the idea!

Keep on rocking.

**Comment by reiieva** [comment]
Hi there, it's been a while. Sorry for not keeping you posted about this, I've been quite busy up until recently.

Around a week or two ago I began working on a new version with your requested features in mind and I'm happy to tell you that it's almost ready for a release, I'm currently just fixing some minor bugs and then all I'll need to do is some extensive testing. Believe me, I have been working for hours every day on rewriting the majority of this mod and updating outdated code incompatible with 1.20.5 and above, it's been really tiring but worth it from my perspective at least. 

I can't give you a specific date but I am planning to release a beta version soon for 1.20.5 and 1.20.6 (and hopefully 1.21 although I haven't done any testing on this version).

Once again, sorry for making you wait so long for any updates on my end.

**Comment by Dumpling00** [comment]
Oh mate this is great to hear, no worries at all!
Looking forward to try out the new mod!!

I should've probably mention though - the modpack we're currently playing on is 1.20.1 - any chance the update will be for this version as well?

Take care!

**Comment by reiieva** [comment]
Ah, I see, unfortunately not. With every Minecraft versions comes major changes to Fabric API due to changes in Minecraft's backend code, I've been coding this whole update with newer versions in mind as I want to keep Strange Berries up to date, therefore as a result the current build I have been working on just wouldn't work with prior versions and compatibility for this new update on older versions of Minecraft would require some serious major changes I just don't have the time to complete.

Sorry!

**Comment by Dumpling00** [comment]
That's a shame. Unfortunately we're playing on a modpack with a lot of features that are only on 1.20.1 so we have no way of moving over to the new version really.
Hope you can reconsider in the future. Thank you for your time.

**Comment by reiieva** [comment]
I'm sorry I couldn't help you here, I've released the update for 1.20.5, 1.20.6 and 1.21.

I'm going to close this issue now, thanks for your continued support of my mod, as I've said, it means the world to me! :)
